packages/yoroi-extension/app/components/settings/categories/general-setting/GeneralSettings.js

```diff
@@ -14,84 +14,113 @@ import FlagLabel from '../../../widgets/FlagLabel';
 
 type Props = {|
   +languages: Array<LanguageType>,
   +currentLocale: string,
-  +onSelectLanguage: {| locale: string |} => PossiblyAsync<void>,
+  +onSelectLanguage: ({| locale: string |}) => PossiblyAsync<void>,
   +isSubmitting: boolean,
   +error?: ?LocalizableError,
 |};
 
+const messages = defineMessages({
+  languageLabel: {
+    id: 'wallet.settings.general.language',
+    defaultMessage: '!!!Language',
+  },
+  languageSelectLabel: {
+    id: 'wallet.settings.general.revamp.languageSelectLabel',
+    defaultMessage: '!!!Select your language',
+  },
+});
+
 @observer
-export default class GeneralSettings extends Component<Props> {
-  static defaultProps: {|error: void|} = {
-    error: undefined
+class GeneralSettings extends Component<Props & InjectedLayoutProps> {
+  static defaultProps: {| error: void |} = {
+    error: undefined,
   };
 
-  static contextTypes: {|intl: $npm$ReactIntl$IntlFormat|} = {
+  static contextTypes: {| intl: $npm$ReactIntl$IntlFormat |} = {
     intl: intlShape.isRequired,
   };
 
-  selectLanguage: string => Promise<void> = async (locale) => {
+  selectLanguage: string => Promise<void> = async locale => {
     await this.props.onSelectLanguage({ locale });
   };
 
   form: ReactToolboxMobxForm = new ReactToolboxMobxForm({
     fields: {
       languageId: {
-        label: this.context.intl.formatMessage(globalMessages.languageSelectLabel),
+        label: this.context.intl.formatMessage(
+          this.props.isRevampLayout
+            ? messages.languageSelectLabel
+            : globalMessages.languageSelectLabel
+        ),
         value: this.props.currentLocale,
-      }
-    }
+      },
+    },
   });
 
   render(): Node {
-    const { languages, isSubmitting, error } = this.props;
+    const { languages, isSubmitting, error, isRevampLayout } = this.props;
     const { intl } = this.context;
     const { form } = this;
     const languageId = form.$('languageId');
     const languageOptions = languages.map(language => ({
       value: language.value,
       label: intl.formatMessage(language.label),
-      svg: language.svg
+      svg: language.svg,
     }));
     const componentClassNames = classNames([styles.component, 'general']);
 
     return (
       <div className={componentClassNames}>
-        <Select
-          labelId="languages-select"
-          {...languageId.bind()}
-          onChange={this.selectLanguage}
-          disabled={isSubmitting}
-          renderValue={value => (
-            <Typography variant="body2">
-              {languageOptions.filter(item => item.value === value)[0].label}
-            </Typography>
-          )}
+        {isRevampLayout && (
+          <Typography variant="body1" mb="16px" color="grayscale.900" fontWeight={500}>
+            {intl.formatMessage(messages.languageLabel)}
+          </Typography>
+        )}
+        <Box
+          sx={{
+            width: isRevampLayout ? '506px' : '100%',
```

Comment:
Better use the prop `maxWidth` to limit width, if not, will cause horizontal scroll for smaller layouts.